### PR TITLE
Enable death tests with HIP for ROCm 5.0.2 and later

### PR DESCRIPTION
**Pull request by dalg24**

### CODE REVIEW

File: core/src/HIP/Kokkos_HIP_Abort.hpp [hunk 1]
```diff
@@ -50,6 +50,18 @@
 
 #include <hip/hip_runtime.h>
 
+// FIXME_HIP ROCm 4.5 version header include would be <rocm/rocm_version.h>
+#if __has_include(<rocm_version.h>)
+#include <rocm_version.h>
+#define KOKKOS_IMPL_ROCM_VERSION \
+  ROCM_VERSION_MAJOR * 10000 + ROCM_VERSION_MINOR * 100 * ROCM_VERSION_PATCH
```
Comment (Rombur) [inline core/src/HIP/Kokkos_HIP_Abort.hpp:57]: This looks wrong. I don't remember exactly but the following makes more sense
```suggestion
  ROCM_VERSION_MAJOR * 10000 + ROCM_VERSION_MINOR * 100 + ROCM_VERSION_PATCH
```

**Review by Rombur — CHANGES_REQUESTED** [review]

File: core/src/HIP/Kokkos_HIP_Abort.hpp (hunk 1 above)
Comment (dalg24) [reply core/src/HIP/Kokkos_HIP_Abort.hpp:57]: Yes it is a typo.  I haven't tested yet.

Comment (dalg24) [reply core/src/HIP/Kokkos_HIP_Abort.hpp]: OK I got around to test it now and it does what I want

**Review by Rombur — APPROVED** [review]

**Review by masterleinad — APPROVED** [review]
I don't like using
```
#if __has_include(<rocm_version.h>)
```
before requiring C++17 too much but it seems to work with the relevant compilers anyway.

**Comment by dalg24** [comment]
> I don't like using
> 
> ```
> #if __has_include(<rocm_version.h>)
> ```
> 
> before requiring C++17 too much but it seems to work with the relevant compilers anyway.

Exactly, my reasoning was both the AMD and the HPE Cray compiler are Clang-based and support that macro.

**Pull request merged**